### DISCUSSION TITLE
allowed_request_methods per strategy

### DISCUSSION THREAD

**Discussion by paulo-il** [omniauth/omniauth#1060] [Ideas]
Hi - I am building a new strategy for OmniAuth that can safely handle GET requests in the request phase. It uses an Open ID Third party initiated flow, and from what I can see, it will not fall into the CVE-2015-9284 Vulnerability.

To make this new strategy accept a GET request in the request phase, I need to do:
```
OmniAuth.config.allowed_request_methods = %i[get post]
```

But I don't want to open that option for other strategies. And I cannot see a way to define that value per strategy on the codebase. 

Has this been discussed before? Any reason why this option is offered as a global config, and not per strategy?

**Comment by BobbyMcWho** [comment]
You'd have to open a PR that adds a strategy option of something like `strategy_allowed_request_methods` that defaults to just %i[post], and everywhere we check something along the lines of `OmniAuth.config.allowed_request_methods.include?(request.request_method.downcase.to_sym)` in the OmniAuth::Strategy class would need to || w/ the new strategy option.

**Reply by paulo-il** [reply]
Thanks for getting back to me @BobbyMcWho. I will give that a try.

**Comment by chap** [comment]
@paulo-il something like this worked for me:
```
OmniAuth.config.allowed_request_methods = [:post, :get]

OmniAuth.config.before_request_phase do |env|
  request = Rack::Request.new(env.dup)

  # POST required except for myProvider strategy
  unless request.post? || env["omniauth.strategy"]&.name.to_s.include?("myProvider")
    raise OmniAuth::AuthenticityError
  end
end
```
